File: app/src/main/java/velkus/moneybagsbudgeting/storage/dao/AccountDao.java
package velkus.moneybagsbudgeting.storage.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

import velkus.moneybagsbudgeting.storage.models.Account;

@Dao
public interface AccountDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void saveAccount(Account account);

    @Delete
    void deleteAccount(Account account);

    @Query("SELECT * FROM account")
    List<Account> getAllAccounts();
}

File: app/src/main/java/velkus/moneybagsbudgeting/storage/dao/AccountDao.kt
package velkus.moneybagsbudgeting.storage.dao

import android.arch.lifecycle.LiveData
import android.arch.persistence.room.*
import velkus.moneybagsbudgeting.storage.models.Account
import velkus.moneybagsbudgeting.storage.models.AccountWithAssociations

@Dao
interface AccountDao {

    @get:Query("SELECT * FROM account")
    val allAccounts: LiveData<List<Account>>

    @get:Query("SELECT * FROM account")
    val allAccountsWithAssociations: LiveData<List<AccountWithAssociations>>

    @Query("SELECT * FROM account WHERE id = :accountId")
    fun getAccountWithAssociations(accountId: Long): LiveData<AccountWithAssociations>

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun saveAccount(account: Account)

    @Delete
    fun deleteAccount(account: Account)
}